### ISSUE TITLE
False positive of `declaration-colon-space-after` for single line on a next line

### ISSUE THREAD

**Issue by AlexWayfer** [stylelint/stylelint#4363]
> Clearly describe the bug

There are false positives for `declaration-colon-space-after` when the value on a next single line. I can't place the value on the line with property name due to line length restriction.

> Which rule, if any, is the bug related to?

`declaration-colon-space-after`

> What CSS is needed to reproduce the bug?

```css
padding:
  1000000px;
```

> What stylelint configuration is needed to reproduce the bug?

```yaml
rules:
  max-line-length:
    - 10
  declaration-colon-space-after:
    - always-single-line

```

> Which version of stylelint are you using?

`11.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI via `stylelint test.css`.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
test.css
 1:9  ✖  Expected single space after ":" with a single-line declaration   declaration-colon-space-after
```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Technically it works as documented:

> There must always be a single space after the colon if the declaration's value is single-line.

Value is a single line, so space is required.

But I understand, that it's not what you would want in this case.

Do you use Prettier on stylesheet that has this problem, by any chance?

**Comment by AlexWayfer** [comment]
> Technically it works as documented:
> 
> > There must always be a single space after the colon if the declaration's value is single-line.

Yes. But I'm in clamp currently: I can't not violate rules. 

> But I understand, that it's not what you would want in this case.

Yes. Maybe, the option can be renamed from `always-single-line` to something like just `single-line`.

> Do you use Prettier on stylesheet that has this problem, by any chance?

Sorry, I didn't understand. If you're talking about https://prettier.io/ — no.

**Comment by jeddy3** [comment]
I don't think there's anything we can do here. The issue is due to the conflict between two rules, and [conflicts can only be resolved by the user as all rules are standalone](https://stylelint.io/user-guide/faq#how-do-i-manage-conflicts-between-rules). 

Any additional option for `declaration-colon-space-after` would render the rule less useful as it would allow writing the following code _even if the maximum line length was greater_:

```css
a {
  padding:
    1000000px;
}
```

This is one of the reasons stylelint [has special command comments](https://stylelint.io/user-guide/configuration#turning-rules-off-from-within-your-css) to turn off rules for specific lines of code. By using these comments you can enforce strict stylistic conventions _for the majority of cases_ while opting out of them in _the occasional edge case_.

```css
a {
  padding: /* stylelint-disable-line declaration-colon-space-after */
    calc(very long value);
}

b {
  padding: 2px; /* This stylistic convention is still enforced */
```

Closing for now. Future viewers are welcome to thumb up the issue, though.

**Comment by AlexWayfer** [comment]
> I don't think there's anything we can do here. The issue is due to the conflict between two rules, and conflicts can only be resolved by the user as all rules are standalone.

Okay… just my opinion: I think, rules can be _related_, like `max-line-length` rule can check if the place about which it found as offense with self-fix will cause an offense from another rule.

Maybe even auto-fix mechanism can be used. `max-line-length` is not auto-fixable, as I see on the site, but `declaration-colon-space-after` is. So… I guess we technically can suppress some offenses for avoiding another offenses and in this way avoid conflicts. Complete independence of rules can hurt (actually, it is already).

**Comment by PhiLhoSoft** [comment]
Actually, I would appreciate an option for this rule allowing to split a line in two after the colon, without necessarily a line length restriction (ie. no relation between two rules), just for readability.
I agree that the given example when the value is short is to avoid, but I doubt many people will abuse it.
On the other hand, I want to be able to write:
```
	background:
		repeating-linear-gradient(
			135deg,
			white 0,
			white 8px,
			salmon 8px,
			salmon 12px
		);
```
or:
```
	box-shadow:
		0 5px 5px -3px rgba(black, 20%),
		0 8px 10px 1px rgba(black, 14%),
		0 3px 14px 2px rgba(black, 12%);
```
just because it is more readable, without having to add a disable comment each time.

Just my two cents.
